pkg/db/v1alpha2/interface.go

```diff
@@ -120,40 +124,526 @@ func New() (KatibDBInterface, error) {
 }
 
 func (d *dbConn) RegisterExperiment(experiment *v1alpha2.Experiment) error {
-	return nil
+	var paramSpecs string
+	var objSpec string
+	var algoSpec string
+	var nasConfig string
+	var start_time string
+	var completion_time string
+	var err error
+	if experiment.ExperimentSpec != nil {
+		if experiment.ExperimentSpec.ParameterSpecs != nil {
+			paramSpecs, err = (&jsonpb.Marshaler{}).MarshalToString(experiment.ExperimentSpec.ParameterSpecs)
+			if err != nil {
+				log.Fatalf("Error marshaling Parameters: %v", err)
+			}
+		}
+		if experiment.ExperimentSpec.Objective != nil {
+			objSpec, err = (&jsonpb.Marshaler{}).MarshalToString(experiment.ExperimentSpec.Objective)
+			if err != nil {
+				log.Fatalf("Error marshaling Objective: %v", err)
+			}
+		}
+		if experiment.ExperimentSpec.Algorithm != nil {
+			algoSpec, err = (&jsonpb.Marshaler{}).MarshalToString(experiment.ExperimentSpec.Algorithm)
+			if err != nil {
+				log.Fatalf("Error marshaling Algorithm: %v", err)
+			}
+		}
+		if experiment.ExperimentSpec.NasConfig != nil {
+			nasConfig, err = (&jsonpb.Marshaler{}).MarshalToString(experiment.ExperimentSpec.NasConfig)
+			if err != nil {
+				log.Fatalf("Error marshaling NasConfig: %v", err)
+			}
+		}
+	}
+	if experiment.ExperimentStatus != nil {
+		s_time, err := time.Parse(time.RFC3339Nano, experiment.ExperimentStatus.StartTime)
+		if err != nil {
+			log.Printf("Error parsing start time %s: %v", experiment.ExperimentStatus.StartTime, err)
+		}
+		start_time = s_time.UTC().Format(mysqlTimeFmt)
+		c_time, err := time.Parse(time.RFC3339Nano, experiment.ExperimentStatus.CompletionTime)
+		if err != nil {
+			log.Printf("Error parsing completion time %s: %v", experiment.ExperimentStatus.CompletionTime, err)
+		}
+		completion_time = c_time.UTC().Format(mysqlTimeFmt)
+	}
+	_, err = d.db.Exec(
+		`INSERT INTO experiments (
```

Comment:
What about the proposal for an abstract DB interface? https://docs.google.com/document/d/1xAi4V9SCAVZvrwiVc6a3VpK7BXPx-liQbr6YPdqBThs/edit#heading=h.1hri22e2jta

Reply:
@richardsliu I think it is an important proposal. Should I include it to this PR? or make another PR after this?
If I include it, this PR may become too big.

Reply:
Personally, I think it is better to place in another PR.

Reply:
Let's change it in another PR.